src/carma-platform/parse_ros2_bags.py

```diff
@@ -26,6 +26,44 @@ def get_rosbag_options(path, serialization_format="cdr", storage_id="sqlite3"):
 
     return storage_options, converter_options
 
+def get_earliest_timestamp(reader, topic_types):
+    """
+    Get the earliest timestamp from the first message of each topic in the bag.
```

Comment:
It looks like this returns the earliest time out of all topics, not the earliest time of each.

Reply:
Edited

Reply:
Since this is using a sequential reader, could we not just return the timestamp of the first message processed on one of the topics of interest?

Reply:
yeah I think you are right. I think it is safe to think that the first message recorded in the bag is actually the earliest timestamp. Even if it is not accurate, it shouldn't be far off from the truth. So I simplified it